sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosClientBuilder.java

```diff
@@ -730,33 +731,24 @@ public CosmosClientBuilder readRequestsFallbackEnabled(boolean readRequestsFallb
      * to be carefully reviewed and tests - which is wht retries for patch can only be enabled on request options
      * - any CosmosClient wide configuration will be ignored.
      * <br/>
-     * Bulk/Delete by PK/Transactional Batch/Stroed Procedure execution: No automatic retries are supported.
-     * @param nonIdempotentWriteRetriesEnabled  a flag indicating whether the SDK should enable automatic retries for
-     * an operation when idempotency can't be guaranteed because for the previous attempt a request has been sent
-     * on the network.
-     * @param useTrackingIdPropertyForCreateAndReplace a flag indicating whether write operations can use the
-     * trackingId system property '/_trackingId' to allow identification of conflicts and pre-condition failures due
-     * to retries. If enabled, each document being created or replaced will have an additional '/_trackingId' property
-     * for which the value will be updated by the SDK. If it is not desired to add this new json property (for example
-     * due to the RU-increase based on the payload size or because it causes documents to exceed the max payload size
-     * upper limit), the usage of this system property can be disabled by setting this parameter to false. This means
-     * there could be a higher level of 409/312 due to retries - and applications would need to handle them gracefully
-     * on their own.
+     * Bulk/Delete by PK/Transactional Batch/Stored Procedure execution: No automatic retries are supported.
+     * @param options the options controlling whether non-idempotent write operations should be retried and whether
+     * trackingIds can be used.
      * @return the CosmosItemRequestOptions
      */
-    public CosmosClientBuilder nonIdempotentWriteRetryPolicy(
-        boolean nonIdempotentWriteRetriesEnabled,
-        boolean useTrackingIdPropertyForCreateAndReplace) {
+    public CosmosClientBuilder nonIdempotentWriteRetryPolicy(NonIdempotentWriteRetryOptions options) {
```

Comment:
Isn't this a breaking change? Am I missing something?

Reply:
It was only opened up in the custom serialization PR. Not released yet.

Reply:
Correct

Reply:
Cool, thanks :)